src/common/spectral-utils.ts

```diff
@@ -0,0 +1,22 @@
+import { Configuration } from './configuration'
+import { AGENT_LAST_UPDATE_DATE } from './constants'
+import { PersistenceContext } from './persistence-context'
+
+export const shouldUpdateSpectralAgent = (): boolean => {
+  return (
+    isOneWeekPassedSinceLastUpdate() &&
```

Comment:
i'd make this more general like `isOverUpdateThreshold` and then if we decide to make this configurable to the user the change would only be in one place